service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java

```diff
@@ -0,0 +1,185 @@
+              .getLandingZone(awsCloudContext, requestedRegion)
+              .orElseThrow(
+                  () -> {
+                    throw new BadRequestException(
+                        String.format(
+                            "Unsupported AWS region: '%s'.", creationParameters.getRegion()));
+                  });
+
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(
+          String.format("Invalid AWS region: '%s'.", creationParameters.getRegion()));
+    }
+
+    logger.info(
+        "createAwsStorageFolder workspace: {}, bucketName: {}, prefix {}, region: {}",
+        workspaceUuid.toString(),
+        landingZone.getStorageBucket().name(),
+        commonFields.getName(),
+        creationParameters.getRegion());
+
+    ControlledAwsStorageFolderResource resource =
+        ControlledAwsStorageFolderResource.builder()
+            .common(commonFields)
+            .bucketName(landingZone.getStorageBucket().name())
+            .prefix(commonFields.getName())
+            .build();
+
+    final ControlledAwsStorageFolderResource createdBucket =
+        controlledResourceService
+            .createControlledResourceSync(
+                resource, commonFields.getIamRole(), userRequest, body.getAwsStorageFolder())
+            .castByEnum(WsmResourceType.CONTROLLED_AWS_STORAGE_FOLDER);
+
+    ApiCreatedControlledAwsStorageFolder response =
+        new ApiCreatedControlledAwsStorageFolder()
+            .resourceId(createdBucket.getResourceId())
+            .awsStorageFolder(createdBucket.toApiResource());
+    return new ResponseEntity<>(response, HttpStatus.OK);
+  }
+
+  @Override
+  public ResponseEntity<ApiAwsStorageFolderResource> getAwsStorageFolder(
+      UUID workspaceUuid, UUID resourceId) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    ControlledAwsStorageFolderResource resource =
+        controlledResourceMetadataManager
+            .validateControlledResourceAndAction(
+                userRequest,
+                workspaceUuid,
+                resourceId,
+                SamConstants.SamControlledResourceActions.READ_ACTION)
+            .castByEnum(WsmResourceType.CONTROLLED_AWS_STORAGE_FOLDER);
+    return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
+  }
+
+  @Override
+  public ResponseEntity<Void> deleteAwsStorageFolder(UUID workspaceUuid, UUID resourceId) {
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    controlledResourceMetadataManager.validateControlledResourceAndAction(
+        userRequest,
+        workspaceUuid,
+        resourceId,
+        SamConstants.SamControlledResourceActions.DELETE_ACTION);
+    logger.info(
+        "deleteAwsStorageFolder workspace: {}, resourceId: {}",
+        workspaceUuid.toString(),
+        resourceId.toString());
+
+    controlledResourceService.deleteControlledResourceSync(workspaceUuid, resourceId, userRequest);
```

Comment:
I expected folder deletion to be async, as object deletion could be a long running operation.

Reply:
Done, changed